lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java

```diff
@@ -4055,6 +4052,23 @@ protected boolean makeLink(UIContainer container, String ID, SimplePageItem i, S
 		return available;
 	}
 
+	
+	private String getLinkText(String linkText, String sakaiId ) {
+		//Create a link with open and due dates for assignments links in Lessons
+		SecurityAdvisor yesMan = (String arg0, String arg1, String agr2) -> SecurityAdvisor.SecurityAdvice.ALLOWED;
+		securityService.pushAdvisor(yesMan);
+		try {
+			AssignmentEntity assignment = (AssignmentEntity) assignmentEntity.getEntity(sakaiId, simplePageBean);		
+	 		df = DateFormat.getDateTimeInstance(DateFormat.DEFAULT, DateFormat.DEFAULT, new ResourceLoader().getLocale());
+	 		linkText += " " + messageLocator.getMessage("simplepage.assignment.open_close_date").replace("{0}", 
+					df.format(assignment.getOpenDate())).replace("{1}", df.format(assignment.getDueDate()));
```

Comment:
You don't need to call `.replace()` twice, just use the `messageLocator.getMessage()` method that takes an array of substitution values to replace, like it was doing before you moved the code.